### ISSUE TITLE
`container::delete-objects` semantics

### ISSUE THREAD

**Issue by itowlson** [WebAssembly/wasi-blobstore#30]
Is the container `delete-objects` function intended to offer any guarantees about atomicity?  If not, what behaviour is expected on partial failure?

**Comment by Mossaka** [comment]
I do not believe `delete-objects` offers any guarantees about atomicity. 

> If not, what behaviour is expected on partial failure?

I think we should clarify this in the docs of this operation. In particular, I do believe it would be helpful to return a list of keys that are deleted successfully and an error message indicating that there is a partial failure, similar to the semantics of [DeleteObjects](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html) in s3.

Interestingly, azure blob store does not support delete-objects operation. 

CC @thomastaylor312

**Comment by itowlson** [comment]
> Interestingly, azure blob store does not support delete-objects operation.

Indeed... I (rather foolishly) picked Azure as my first try to implement wasi-blobstore and ran slap bang into this!  There is a "blob batch" API but the SDK for my language doesn't surface it.

`container::clear()` seems not to exist in Azure either.

**Comment by thomastaylor312** [comment]
I really like the idea of adding an enum variant type thing to the error of deleted object names (or maybe return the list every time, success or error). That could help with failures
